### ISSUE TITLE
Add metadata constants to skills

### ISSUE THREAD

**Issue by jacobtomlinson** [opsdroid/opsdroid#121]
Skill developers should be able to write a couple of string constants describing their skill and it's usage, these should be stored somewhere in opsdroid when the modules are loaded. An official "help" skill should also be created to allow users to access these constants.

**Comment by nicgrayson** [comment]
TL;DR: I want to write to code required for this issue

I'm looking to switch to a python based chatbot away from [lita](http://lita.io). It's a huge difference to not have a help that lists all commands you can send the bot. I've thought about a few ways to make this work but they each change how skills are created a fair amount.
1. Add a parameter to the matchers like `def match_regex(regex, case_sensitive=True, usage="")` then collect all of the usage statements:
```python
@match_regex(r'help$', usage='Displays this message')
async def help(opsdroid, config, message):
```
2. Add a new decorator for the usage of the skill:
```python
@match_regex(r'help$')
@opsdroid.usage('Displays this message')
async def help(opsdroid, config, message):
```
3. Use the `__doc__` field on the function:
```python
@match_regex(r'help$')
async def help(opsdroid, config, message):
    """Displays this help message"""
```

The problem with either of these is getting the skill name to display. Right now the help function I'm writing displays like this because it pulls from the skill name:
```
help: Display usage for provided skill
help: Displays this message
```
This should be something like this IMO:
```
help: Display usage for provided skill
help [skill_name]: Displays this message
```

To solve this second problem we could:
1. Add a decorator of the command name then concat the name with the usage:
```python
@match_regex(r'help (.*)')
@usage('Displays this message')
@command_name('help [skill_name]')
async def help_skill(opsdroid, config, message):
```
2. Add the name of the skill to the usage:
```python
@match_regex(r'help (.*)')
async def help_skill(opsdroid, config, message):
    """help [skill_name]: Displays usage for provided skill"""
```
I'm thinking adding the skill name and human readable pattern to the `__doc__`, usage decorator, or matcher input make the most sense.

Let me know which way you want to go and I'll send a PR for this project and a bunch of skills 😄  I will also write a help skill. Let me know if you want me to just transfer that in or if you want to create a black project in the opsdroid org for me to PR. If you want to chat about this I can hop in gitter or skype.

**Comment by jacobtomlinson** [comment]
@nicgrayson awesome we would love you to contribute!

I hadn't even considered using the docstrings to do it but I think that is a really neat idea. Let's go with that. Not sure the best way to solve the second problem though, neither option is appealing. I wonder if you could use the function name somehow?

In terms of the help skill if you create it in your own GitHub and then we can transfer it over to the opsdroid org when it's done. 

I think @FabioRosado had a look at something similar a while back so he might have some thoughts.

**Comment by jacobtomlinson** [comment]
@nicgrayson feel free to drop into the developer room on the [gitter](https://gitter.im/opsdroid) too.

**Comment by nicgrayson** [comment]
I've created a help skill that pulls from docstrings: https://github.com/nicgrayson/skill-help

**Comment by jacobtomlinson** [comment]
Sorry for the delay, I missed this comment. 

The skill looks great. If you want to transfer the ownership of the repository to me I'll move it into the opsdroid org and it will become an official skill.

It might also be worth putting in a PR to add it to the example configuration as an automatically enabled skill.

**Comment by nicgrayson** [comment]
@jacobtomlinson i set a transfer to you then you can put it in the org. I'll send a docs PR soon.

**Comment by jacobtomlinson** [comment]
Transferred! Many thanks. 

Ping me your address via DM on [Twitter](https://twitter.com/_jacobtomlinson) or [Gitter](https://gitter.im/opsdroid) and I'll post you some [opsdroid stickers](https://medium.com/opsdroid/stickers-for-contributors-a0a1f9c30ec1)!
